src/toil/jobStores/abstractJobStore.py

```diff
@@ -100,6 +100,10 @@ def __init__(self, locator):
             "the job store with 'toil clean' to start the workflow from scratch" % locator)
 
 
+class JobStoreBucketExistsException(Exception):
```

Comment:
Shouldn't the name be `BucketLocationConflictException`? We do handle existing buckets, just not in the wrong region.